pkg/apis/coordination/types.go

```diff
@@ -20,6 +20,22 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type CoordinatedStrategy string
```

Comment:
I think that we should use the public types instead of replicate them internally and avoid them drifting

Reply:
ack, fixed.

Reply:
@aojea @thockin: I'm a bit confused at how to align the internal and external types. Should Lease.Strategy have type v1.CoordinatedLeaseStrategy? Should an alias be defined in the internal types `type CoordinatedLeaseStrategy v1.CoordinatedLeaseStrategy`? 

Tried to look for prior examples but must of them redefine the types in internal
eg: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L819-L839
 https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/types.go#L107-L168

Reply:
There might be examples where we alias versioned types to internal types (but never the other way around!).. but I think in general we want to avoid that and define all types for internal.  Someone please correct me if there is some reason why we would want to type share.

Reply:
In a perfect world, we would define things one time.  Go is not a perfect world and I don't think we do this anywhere else.  MNostly we just double-define types and consts.

We know of plenty of places where the internal type differs from external.

In theory, the internal consts could, too, but I don't think conversion-gen handles that :)

e.g. it could be "OldestCompatVersion" in core/v1, "EldestCompatVersion" in core/v2, and "OldestVersion" in internal, and as long as we convert it, it will pass roundrtip.  I don't know WHY we would do that, but we could.

```
$ git grep ProtocolTCP.*= '*api*/types.go'
pkg/apis/core/types.go: ProtocolTCP Protocol = "TCP"
staging/src/k8s.io/api/core/v1/types.go:        ProtocolTCP Protocol = "TCP"

$ git grep "type Protocol " '*api*/types.go'
pkg/apis/core/types.go:type Protocol string
staging/src/k8s.io/api/core/v1/types.go:type Protocol string
```